### ISSUE TITLE
Register a new access-type with IANA

### ISSUE THREAD

**Issue by Pandapip1** [ipfs/specs#310]
I suggest that a new access type be created for IPFS multihashes, as per RFC 1341 section 7.3.3

```
Content-type: message/external-body; access-type=ipfs
```

This would allow for websites to directly serve files off of IPFS.

**Comment by lidel** [comment]
Mind elaborating @Pandapip1 ? What use cases are solved by this?

**Comment by Pandapip1** [comment]
Uh... primarily being able to serve IPFS multihashes natively, without storing their contents or serving them. This might be particularly useful when combined with e.g. https://eips.ethereum.org/EIPS/eip-5219

**Comment by Pandapip1** [comment]
I've decided that I'll do this myself: https://github.com/Pandapip1/ipfs-access-type-internet-draft. Anyone should feel free to contribute!

**Comment by Pandapip1** [comment]
I'm switching this to be multiaddresses in general.

**Comment by Pandapip1** [comment]
The utility behind `message/external-body` is to have a lightweight server (such as [might be run on the blockchain](https://eips.ethereum.org/EIPS/eip-5219)) tell web browsers/whatever is receiving the response to go somewhere else to get the full data.

**Comment by Pandapip1** [comment]
Updated the main issue to bring it more up to date

**Comment by Pandapip1** [comment]
Would prefer if this stayed open.

**Comment by Pandapip1** [comment]
> Problem 1: The proposal uses HTTP multiaddrs for some reason, does not even use CIDs – how is it related to IPFS?

It used to use CIDs, now it uses multiaddrs. Unless there's a better place for this issue, the ipfs/specs repository is the correct place.

> Problem 2: Anyway, this still feels to me like reinventing HTTP redirect with extra steps – an unnecessary complication, because you can already do all this today (without inventing anything new):

First of all: this doesn't reinvent the wheel. The `message/external-body` MIME type already exists. All this proposal is add a new access type (in a similar manner to https://datatracker.ietf.org/doc/html/rfc2017).

The reason for this over 301/308 redirects is this *doesn't* redirect. If you're hosting `https://mywebsite.com/ab/cd` using this access type, you keep the `https` URI scheme, `mywebsite.com` domain, `https://mywebsite.com` origin, could link to `/ef/gh`, `../hi`, and `//example.com`, etc...

TL;DR: The rationale for `message/external-body` in general applies here.

> Problem 3: The proposal does not explain value added by inventing a special response type here. Introducing complexity like this makes it harder for people to adopt content addressing, if you propose something it should be clear what value is added and why it could not be done via preexisting means.

Oops, I forgot to include that. Thanks for bringing that up!
